// @microsoft/sp-property-pane/lib-commonjs/common/KillSwitches.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.styleChoiceGroupUsingCustomCSSKSActivated = exports.removeChoiceGroupCustomCSSKSActivated = exports.highContrastPropertyPaneGroupButtonKSActivated = exports.choiceGroupOptionTextOverflowKSActivated = exports.isAllowEmptyPropertyNameKSActivated = exports.isUpdatePropertyPaneAriaLabelKSActivated = exports.isFixEmptyAlertAndFocusKSActivated = void 0;
var sp_core_library_1 = require("@microsoft/sp-core-library");
var KillSwitches = /** @class */ (function () {
    function KillSwitches() {
    }
    KillSwitches.isSubPropertyLabelOverrideKillSwitchActivated = function () {
        return sp_core_library_1._SPKillSwitch.isActivated(sp_core_library_1.Guid.parse('34362b85-342e-4a2d-9966-db50c2e09976'), '02/27/2020', 'Support sub property dropdown label override.');
    };
    KillSwitches.textFieldOnBlurValidateKsActivated = function () {
        return sp_core_library_1._SPKillSwitch.isActivated('4b094488-064a-433d-bcf3-8d0f31eeb668'
        /* '11/30/2020', 'Fix TextField empty string on blur' */
        );
    };
    KillSwitches.isChoiceGroupRTLmodeKillSwitchActivated = function () {
        /* '2020-12-10', 'Enable choice group svg and label normal in RTL mode' */
        return sp_core_library_1._SPKillSwitch.isActivated('ff93175a-a652-4748-bb9d-bdeb16cbf196');
    };
    KillSwitches.isRemoveFixedTextHeightKillSwitchActivated = function () {
        return sp_core_library_1._SPKillSwitch.isActivated('0b2cf2c8-3b01-46b4-8324-aec1e344a738'
        /* '2021-03-30', 'Remove fixed text height for choice group' */
        );
    };
    KillSwitches.expandAlternateFieldsConnectorMarginKSActivated = function () {
        return sp_core_library_1._SPKillSwitch.isActivated('9f84a18b-fb44-4cdd-86d1-7fe86bde2780'
        /* '2021-05-28', 'expand margin bottom for alternate fields connector' */
        );
    };
    return KillSwitches;
}());
exports.default = KillSwitches;
function isFixEmptyAlertAndFocusKSActivated() {
    return sp_core_library_1._SPKillSwitch.isActivated('6b300f30-5d20-4f5a-acbf-8ec1527473f7'
    /* '06/02/2021', 'Fix empty alert and make focus fall back to whole property pane if there is no focusable element in property pane content.' */
    );
}
exports.isFixEmptyAlertAndFocusKSActivated = isFixEmptyAlertAndFocusKSActivated;
function isUpdatePropertyPaneAriaLabelKSActivated() {
    return sp_core_library_1._SPKillSwitch.isActivated('9c35062b-5952-4959-97bc-d0d19ad99dc0'
    /* '06/18/2021', 'update property pane aria label.' */
    );
}
exports.isUpdatePropertyPaneAriaLabelKSActivated = isUpdatePropertyPaneAriaLabelKSActivated;
function isAllowEmptyPropertyNameKSActivated() {
    return sp_core_library_1._SPKillSwitch.isActivated('d39e74cc-8615-40e7-9f1b-bd6ad16dda59'
    /* '07/08/2021', 'Allow but do not persist properties with empty property name' */
    );
}
exports.isAllowEmptyPropertyNameKSActivated = isAllowEmptyPropertyNameKSActivated;
function choiceGroupOptionTextOverflowKSActivated() {
    return sp_core_library_1._SPKillSwitch.isActivated('7454c194-0ee3-40ad-8de8-b37e6b1ab9c1'
    /* '09/22/2021', 'Fixes issue related to text that overflows with Text Spacing on in the Choice Group Option component' */
    );
}
exports.choiceGroupOptionTextOverflowKSActivated = choiceGroupOptionTextOverflowKSActivated;
function highContrastPropertyPaneGroupButtonKSActivated() {
    return sp_core_library_1._SPKillSwitch.isActivated('eb0e4532-4690-471d-b63e-895bac23117a'
    /* '09/21/2021', 'Makes the selected state for property pane group button more obvious' */
    );
}
exports.highContrastPropertyPaneGroupButtonKSActivated = highContrastPropertyPaneGroupButtonKSActivated;
function removeChoiceGroupCustomCSSKSActivated() {
    return sp_core_library_1._SPKillSwitch.isActivated('cb38ec07-5336-4982-8185-8fd7790166d5'
    /* '10/04/2021', 'Remove custom CSS styles for ChoiceGroup' */
    );
}
exports.removeChoiceGroupCustomCSSKSActivated = removeChoiceGroupCustomCSSKSActivated;
function styleChoiceGroupUsingCustomCSSKSActivated() {
    return sp_core_library_1._SPKillSwitch.isActivated('bd838357-d9ed-4798-b68c-ea458d17ae91'
    /* '10/18/2021', 'Restore a subset of custom CSS styles for ChoiceGroup removed by KS cb38ec07-5336-4982-8185-8fd7790166d5 and fix line height and text width issue' */
    );
}
exports.styleChoiceGroupUsingCustomCSSKSActivated = styleChoiceGroupUsingCustomCSSKSActivated;
